java/rocksjni/writebatchhandlerjnicallback.cc

```diff
@@ -83,6 +88,92 @@ bool WriteBatchHandlerJniCallback::Continue() {
   return static_cast<bool>(jContinue == JNI_TRUE);
 }
 
+void WriteBatchHandlerJniCallback::SingleDelete(const Slice& key) {
+  const jbyteArray j_key = sliceToJArray(key);
+
+  m_env->CallVoidMethod(
+                        m_jWriteBatchHandler,
+                        m_jSingleDeleteMethodId,
+                        j_key);
+
+  m_env->DeleteLocalRef(j_key);
+}
+
+Status WriteBatchHandlerJniCallback::PutCF(uint32_t column_family_id, const Slice& key,
+                                             const Slice& value) {
+  if (column_family_id == 0) {
+    Put(key, value);
+  } else {
+    const jbyteArray j_key = sliceToJArray(key);
+    const jbyteArray j_value = sliceToJArray(value);
+
+    m_env->CallVoidMethod(
+                          m_jWriteBatchHandler,
+                          m_jPutCFMethodId,
+                          column_family_id,
+                          j_key,
+                          j_value);
+
+    m_env->DeleteLocalRef(j_value);
+    m_env->DeleteLocalRef(j_key);
+  }
+  return Status::OK();
```

Comment:
Right, I wondered about that too. There seems a few things at play here:
    * Ideally the method return type should be "void".  However the C++ the "PutCF/GetCF" RocksDB required a "Status" return type which is different from the default "Put/Get/...". Not sure about the original design intent there. But I have to keep the method signature to make the C++ compiler happy. 
     * we could add some status return type on the Java callback side (currently it's void as well). But I am not sure what purpose the "status" return type could serve. 
     * Exception. Assume Java side throws exception, we could stuff the exception message in the Status code. But I am not sure if RocksDB cared, from the attached inlined message, it seems returning S_OK is just as good. The JVM JNI could force the exception to the Java side anyhow. So the app code on the Java side will need to handle the exception.  

Please let me know what you think. 

====
virtual Status PutCF(uint32_t column_family_id, const Slice& key,
                         const Slice& value) {
      if (column_family_id == 0) {
        // Put() historically doesn't return status. We didn't want to be
        // backwards incompatible so we didn't change the return status
        // (this is a public API). We do an ordinary get and return Status::OK()
        Put(key, value);
        return Status::OK();
      }
      return Status::InvalidArgument(
          "non-default column family and PutCF not implemented");
    }